applications/graphQL/query.py

```
@@ -0,0 +1,96 @@
+        """Resolve a random active insult, optionally filtered by category."""
+        queryset = Insult.objects.filter(status="A")
+        if category:
+            queryset = queryset.filter(category=category)
+        return queryset.order_by("?").first()
+
+    def resolve_insult_by_id(root, info, id: str) -> Insult:
+        """Resolve an insult by its ID."""
+        try:
+            return Insult.objects.get(id=id)
+        except ObjectDoesNotExist:
+            raise GraphQLError(f"Insult with ID {id} not found")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)